platformio_esp32_envs.ini

```diff
@@ -52,4 +52,31 @@ debug_tool                = ftdi
 debug_extra_cmds          = break Misc.ino:3011
 extra_scripts             = ${esp32_common.extra_scripts}
 
+; Custom: 4096k version --------------------------
+[env:custom_ESP32_4M316k_ETH]
+extends                   = esp32_common
+platform                  = ${esp32_common.platform}
+build_flags               = ${esp32_common.build_flags}   -DPLUGIN_BUILD_CUSTOM -DHAS_ETHERNET
+board                     = esp32dev
+extra_scripts             = ${esp32_common.extra_scripts}
+                            pre:pre_custom_esp32.py
+
+[env:test_ESP32_4M316k_ETH]
+extends                   = esp32_common
+platform                  = ${esp32_common.platform}
+build_flags               = ${esp32_common.build_flags}  -DPLUGIN_SET_TEST_ESP32 -DHAS_ETHERNET
+board                     = esp32dev
+extra_scripts             = ${esp32_common.extra_scripts}
+
+
+[env:test_ESP32-wrover-kit_4M316k_ETH]
+extends                   = esp32_common
+platform                  = ${esp32_common.platform}
+build_flags               = ${esp32_common.build_flags}  -DPLUGIN_SET_TEST_ESP32 -DHAS_ETHERNET
+board                     = esp-wrover-kit
+upload_protocol           = ftdi
+debug_tool                = ftdi
+debug_extra_cmds          = break Misc.ino:3011
+extra_scripts             = ${esp32_common.extra_scripts}
```

Comment:
These environments can be much more simple.
The only thing you want to change is the `build_flags`, so then do that.

For example:

```ini
[env:test_ESP32-wrover-kit_4M316k_ETH]
extends                   = test_ESP32-wrover-kit_4M316k
build_flags               = ${test_ESP32-wrover-kit_4M316k.build_flags}  -DHAS_ETHERNET
```

Reply:
Yes, thats right of course, I change it.

Reply:
Didn't get it. When I do these changes, I do not see the build option in platformio anymore

Reply:
How did you change it?

Reply:
This didn't work
[env:test_ESP32-wrover-kit_4M316k_ETH]
extends                   = env:test_ESP32-wrover-kit_4M316k
build_flags               = ${env:test_ESP32-wrover-kit_4M316k.build_flags} -DHAS_ETHERNET

This works
[env:test_ESP32-wrover-kit_4M316k_ETH]
extends                   = env:test_ESP32-wrover-kit_4M316k
platform                  = ${env:test_ESP32-wrover-kit_4M316k.platform}
build_flags               = ${env:test_ESP32-wrover-kit_4M316k.build_flags} -DHAS_ETHERNET

It seems, tha platform is mandatory. I don't know why. I changed it and pushed it to my branch
https://github.com/kretzp/ESPEasy/tree/feature/ethernet_support